drivers/soundwire/intel.c

```diff
@@ -1412,6 +1412,96 @@ static int intel_master_remove(struct sdw_master_device *md)
 
 #ifdef CONFIG_PM
 
+static int _suspend(struct device *dev, bool clock_stop)
+{
+	struct sdw_cdns *cdns = dev_get_drvdata(dev);
+	struct sdw_intel *sdw = cdns_to_intel(cdns);
+	int ret;
+
+	dev_err(dev, "%s start\n", __func__);
+
+	if (clock_stop) {
+		ret = sdw_cdns_suspend(cdns, true);
+		if (ret < 0) {
+			dev_err(dev, "cannot enable clock stop on suspend\n");
+			return ret;
+		}
+	}
+
+	ret = sdw_cdns_enable_interrupt(cdns, false);
+	if (ret < 0) {
+		dev_err(dev, "cannot disable interrupts on suspend\n");
+		return ret;
+	}
+
+	ret = intel_link_power_down(sdw);
+	if (ret) {
+		dev_err(dev, "Link power down failed: %d", ret);
+		return ret;
+	}
+
+	/* set WAKEEN interrupt for wake-up events */
+	intel_shim_wake(sdw, clock_stop);
+
+	dev_err(dev, "%s done\n", __func__);
```

Comment:
same comment for all dev_err's in this patch

Reply:
thanks! I will refine it.